scripts/check_patch_checks.py

```diff
@@ -0,0 +1,71 @@
+import argparse
+import requests
+import json
+import os
+from collections import defaultdict
+from typing import Dict, List, Set
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Create Patch Files")
+    parser.add_argument(
+        "-start",
+        "--start",
+        metavar="<string>",
+        type=str,
+        help="Start timestamp for patches",
+    )
+    parser.add_argument(
+        "-end",
+        "--end",
+        metavar="<string>",
+        type=str,
+        help="End timestamp for patches",
+    )
+    return parser.parse_args()
+
+def make_api_request(url):
+    print(url)
+    r = requests.get(url)
+    return json.loads(r.text)
+
+def check_patch(patch):
+    url = patch['checks']
+    checks = [check for check in make_api_request(url) if check["user"]["username"] == "rivoscibot"]
+
+    # first three checks should be lint start/finish and apply
+    if len(checks) < 3:
+        return True
+
+    apply_failure = False
+    testsuite_reported = False
+    
+    for check in checks:
+        if check["description"] == "Patch failed to apply":
+            apply_failure = True
+        if check["context"] == "toolchain-ci-rivos-test":
+            testsuite_report = True
+
+    # If testsuite was not reported, mark for rerun
+    # testsuite should be reported as long as it was applied
+    if not apply_failure and not testsuite_reported:
+        return True
+
+    return False
+
+
+def get_patches(start: str, end: str):
+    url = f"https://patchwork.sourceware.org/api/1.3/patches/?order=date&q=RISC-V&project=6&since={start}&before={end}"
```

Comment:
This looks good as long as it's invoked with old ( > 6hr old) timestamps. 
Newer stuff won't have testsuite reports.
That way if github doesn't run the scheduled run 4 times in a row we process a patch twice... not ideal but that's fine.

Reply:
I'm not sure if we should run this just one day behind or two days behind. If it's two days behind, we don't need to worry about duplicate patches but the latency increases.